core/src/main/java/org/elasticsearch/action/support/replication/ReplicationRequest.java

```diff
@@ -42,7 +42,12 @@
 
     public static final TimeValue DEFAULT_TIMEOUT = new TimeValue(1, TimeUnit.MINUTES);
 
-    ShardId internalShardId;
+    /**
+     * Target shard the request should execute on. In case of index and delete requests,
+     * shard id gets resolved by the transport action before performing request operation
+     * and at request creation time for shard-level bulk, refresh and flush requests.
+     */
+    protected ShardId resolvedShardId;
```

Comment:
I like naming this `shardId` instead of `resolvedShardId`, changed in https://github.com/areek/elasticsearch/commit/fd2b09211f73e801b1e8f4bf51ba4f5cc19ec347